diesel/src/expression/expression_methods/text_expression_methods.rs

```diff
@@ -1,8 +1,39 @@
 use expression::{Expression, AsExpression};
-use expression::predicates::{Like, NotLike};
+use expression::predicates::{Concat, Like, NotLike};
 use types::Text;
 
 pub trait TextExpressionMethods: Expression<SqlType=Text> + Sized {
+    /// Concatenates two strings using the `||` operator.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("src/doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #     users {
+    /// #         id -> Integer,
+    /// #         name -> VarChar,
+    /// #     }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     use self::users::dsl::*;
+    /// #     let connection = establish_connection();
+    /// #
+    /// let names = users.select(name.concat(" the greatest")).load(&connection);
+    /// let expected_names = vec![
+    ///     "Sean the greatest".to_string(),
+    ///     "Tess the greatest".to_string(),
```

Comment:
The most important stuff first: In this case "greatest" should be capitalized as it's part of the title "{Sean,Tess} the Greatest" (same as in "Alexander the Great") 😉